generator/parser/binder.cpp

```diff
@@ -331,7 +331,8 @@ void Binder::declare_symbol(SimpleDeclarationAST *node, InitDeclaratorAST *init_
       fun->setVariadics (decl_cc.isVariadics ());
 
       // ... and the signature
-      foreach (DeclaratorCompiler::Parameter p, decl_cc.parameters())
+      const QList<DeclaratorCompiler::Parameter> parameters = decl_cc.parameters();
```

Comment:
Would `const auto` work here? If we are using C++11 as we do, we could really use `auto` a bit more (though there are certainly cases which only work from C++17 onwards).